src/services/OswServiceTaskBLEServer.cpp

```diff
@@ -301,25 +301,19 @@ void OswServiceTaskBLEServer::BatteryLevelStatusCharacteristicCallbacks::onRead(
 
 void OswServiceTaskBLEServer::CurrentTimeCharacteristicCallbacks::onRead(NimBLECharacteristic* pCharacteristic) {
     time_t offset = 0;
-    uint32_t day;
-    uint32_t month;
-    uint32_t year;
-    OswHal::getInstance()->getDate(offset, &day, &month, &year);
-    uint32_t hour;
-    uint32_t minute;
-    uint32_t second;
-    OswHal::getInstance()->getUTCTime(&hour, &minute, &second);
-    uint32_t dow;
-    OswHal::getInstance()->getDate(offset, &day, &dow);
-
-    this->bytesCurrentTime[0] = (uint8_t) year; // Exact Time 256 -> Day Date Time -> Date Time -> Year #1
-    this->bytesCurrentTime[1] = (uint8_t) (year >> 8); // Exact Time 256 -> Day Date Time -> Date Time -> Year #2
-    this->bytesCurrentTime[2] = (uint8_t) month; // Exact Time 256 -> Day Date Time -> Date Time -> Month
-    this->bytesCurrentTime[3] = (uint8_t) day; // Exact Time 256 -> Day Date Time -> Date Time -> Day
-    this->bytesCurrentTime[4] = (uint8_t) hour; // Exact Time 256 -> Day Date Time -> Date Time -> Hours
-    this->bytesCurrentTime[5] = (uint8_t) minute; // Exact Time 256 -> Day Date Time -> Date Time -> Minutes
-    this->bytesCurrentTime[6] = (uint8_t) second; // Exact Time 256 -> Day Date Time -> Date Time -> Seconds
-    this->bytesCurrentTime[7] = (uint8_t) (dow == 0 ? 7 : dow); // Exact Time 256 -> Day Date Time -> Day of Week
+    OswDate oswDate;
+    OswHal::getInstance()->getDate(offset, oswDate);
+    OswTime oswTime;
+    OswHal::getInstance()->getUTCTime(oswTime);
```

Comment:
why not use `getLocalDate/getLocalTime`?

Reply:
If you have such questions, please take a look into the specification first: https://www.bluetooth.com/specifications/specs/current-time-service-1-1/

-> We implement the current time service with the current time field, NOT local time field.